python/lvmdrp/utils/metadata.py

```diff
@@ -0,0 +1,828 @@
+    if expnum is not None:
+        logger.info(f"filtering by {expnum = }")
+        query.append("expnum == @expnum")
+    if exptime is not None:
+        logger.info(f"filtering by {exptime = }")
+        query.append("exptime == @exptime")
+    if neon is not None:
+        logger.info(f"filtering by {neon = }")
+        query.append("neon == @neon")
+    if hgne is not None:
+        logger.info(f"filtering by {hgne = }")
+        query.append("hgne == @hgne")
+    if krypton is not None:
+        logger.info(f"filtering by {krypton = }")
+        query.append("krypton == @krypton")
+    if xenon is not None:
+        logger.info(f"filtering by {xenon = }")
+        query.append("xenon == @xenon")
+    if argon is not None:
+        logger.info(f"filtering by {argon = }")
+        query.append("argon == @argon")
+    if ldls is not None:
+        logger.info(f"filtering by {ldls = }")
+        query.append("ldls == @ldls")
+    if quartz is not None:
+        logger.info(f"filtering by {quartz = }")
+        query.append("quartz == @quartz")
+    if quality is not None:
+        logger.info(f"filtering by {quality = }")
+        query.append("quality == @quality")
+    if stage is not None:
+        logger.info(f"filtering by {stage = }")
+        query.append("stage == @stage")
+    if status is not None:
+        logger.info(f"filtering by {status = }")
+        query.append("status == @status")
+
+    if query:
+        query = " and ".join(query)
+        metadata.query(query, inplace=True)
+
+    return metadata
+
+
+def _create_store(observatory):
+    """return the metadata store given a path
+
+    Parameters
+    ----------
+    observatory: str
+        observatory for which a metadata store will be created
+    """
+    metadata_path = os.path.join(access.base_dir, "metadata.hdf5")
+
+    logger.info(f"creating metadata store '{metadata_path}'")
+    store = h5py.File(metadata_path, mode="w")
+
+    # add observatory group if needed
+    if observatory not in store:
+        store.create_group(f"{observatory}/raw")
+        store.create_group(f"{observatory}/master")
+
+    store.close()
+
+
+def _del_store():
+    """delete the entire HDF store if exists"""
+    metadata_path = os.path.join(access.base_dir, "metadata.hdf5")
+
+    if os.path.isfile(metadata_path):
+        logger.info(f"removing metadata store '{metadata_path}'")
+        os.remove(metadata_path)
+    else:
+        logger.warning(f"no '{metadata_path}' store found")
+
+
+def _load_store(observatory, mode="r"):
+    """return the metadata store given a observatory
+
+    Parameters
+    ----------
+    observatory : str
+        metadata observatory from which a store will be loaded
+    mode : str, optional
+        the mode in which to open the HDF5 store, by default 'r'
+
+    Returns
+    -------
+    h5py.Group
+        the metadata store for the given observatory
+    """
+    metadata_path = os.path.join(access.base_dir, "metadata.hdf5")
```

Comment:
The base dir in sdss_access is the root directory of the SAS.  LVM can't write to this directory.  Instead, this file should live inside LVM's reduction directory at $LVM_SPECTRO_REDUX/{drpver}/.  

Depending on the size of this file, and how it grows over time, it may make more sense to split this file up by MJD, and store it inside the the [tileid]/[mjd] subdirectories.  Would that affect how you use this file?

Reply:
Ok, great.  I'll test this as well.